lib/src/repository/check_repository.dart

```diff
@@ -87,12 +87,18 @@ Future<VerifiedRepository?> checkRepository({
       }
 
       Future<_PubspecMatch> matchRepoPubspecYaml(String path) async {
+        late String content;
         // checkout the pubspec.yaml at the assumed location
-        final content = await repo.showStringContent(
-          branch!,
-          path,
-          maxOutputBytes: _maxPubspecBytes,
-        );
+        try {
+          content = await repo.showStringContent(
+            branch!,
+            path,
+            maxOutputBytes: _maxPubspecBytes,
+          );
+        } on ArgumentError catch (e) {
```

Comment:
Catching an (Argument) Error is almost always wrong.

Seems to me there is no good way for the caller to verify the path up front, so passing a bad path should not be an error.

Maybe `repo.showStringContent` should be throwing an `Exception` instead in that case or maybe return null to signify a failed attempt?

Reply:
Done (using GitToolException).